### DISCUSSION TITLE
Question on GitHub Workspaces and Spark

### DISCUSSION THREAD

**Discussion by gmoore-liv** [community/community#142997] [Copilot Conversations]
We use Azure DevOps for source control, pull requests, test plans, CI/CD, etc.
We also use GitHub Copilot.

Would we have to migrate/move all of our code to GitHub repositories to utilize GitHub Workspaces and Spark using our existing codebases (instead of using DevOps)?

**Comment by ianshulx** [comment]
You don't have to move all your code from Azure DevOps to GitHub to take advantage of GitHub Workspaces and Spark. You can keep using Azure DevOps for CI/CD while integrating GitHub for collaboration and project management. This hybrid approach allows you to use Azure Boards with GitHub repositories, enabling better tracking of work items. If you do decide to migrate, GitHub provides tools to help import your repositories from Azure DevOps easily. This way, you can enjoy the benefits of both platforms without having to completely overhaul your existing setup

**Comment by gmoore-liv** [comment]
"You can keep using Azure DevOps for CI/CD while integrating GitHub for collaboration and project management."

Thank you! How would I go about that integration?

Possibly I'm slightly jumping the gun and need to wait for this week's releases from GitHub?

**Reply by ianshulx** [reply]
check out this video: https://youtu.be/CYAYomx00PA?si=Ycx4EpfyelU654xY

**Reply by gmoore-liv** [reply]
The YouTube video linked above explained how to connect GitHub repos with Devops boards. That isn't exactly what I'm wanting.

I'm looking for a way to use GitHub features like Copilot AI in codespaces, Copilot AI in PRs but continue to keep our code in Azure Devops.

I'm thinking the answer is to wait until Microsoft releases similar features in Azure Devops.

Migrating all of our code to GitHub would be a massive undertaking.

**Comment by kyleanthony8** [comment]
I am also confronted with this as well, any idea how this could work? Maintaining a codebase in Azure for CI/CD but also a version on Github for exploring AI Features such as Spark?

**Comment by AmanRai8** [comment]
No, you don’t have to migrate all your code to GitHub to use GitHub Copilot or GitHub Workspaces.

- GitHub Copilot works in your code editor (like VS Code) regardless of where your repo is hosted — including Azure DevOps repos.

- GitHub Workspaces and Spark features are tightly integrated with GitHub repos, so to fully use them, your code would need to be on GitHub.

- But you can keep using Azure DevOps for source control, CI/CD, etc., and still get Copilot’s AI help locally.

So, migration is only necessary if you want full GitHub-native features like Workspaces and Spark on your repos.

**Comment by anas-abujaber** [comment]
Yes, to fully utilize GitHub Workspaces and GitHub Copilot for Spark with your existing codebases, your code needs to be in GitHub repositories. GitHub Workspaces and Spark are deeply integrated into the GitHub ecosystem, and do not support Azure DevOps repositories directly.
